### PR TITLE
[rocprofiler-compute] Add test-torch-trace-collector to CI

### PR DESCRIPTION
**Pull request by ggottipa-amd**
## Motivation

`test-torch-trace-collector` is registered as a ctest but is not available on CI. This PR enrolls it alongside the other C++ gtests.

## Technical Details

- Run test-torch-trace-collector through tests/torch_trace_collector_launcher.py so lib dirs are resolved at test time.
- Keep add_test() in the top-level CMakeLists.txt and drop the hardcoded LABELS (YAML labels are directory-scoped).
- Registered only when the torch target exists.

## JIRA ID

AIPROFCOMP-716

## Test Plan

`ctest -N -L full -LE full_rocprofiler_compute_ci_exclude |grep test-torch-trace-collector`

## Test Result
image

## Submission Checklist

- [ ] Look over the contributing guidelines at https://github.com/ROCm/rocm-systems/blob/develop/CONTRIBUTING.md.

### CODE REVIEW

File: projects/rocprofiler-compute/src/lib/roctx_recordfn/tests/CMakeLists.txt [hunk 1]
```diff
@@ -101,7 +101,7 @@ add_test(NAME ${target} COMMAND ${_rcfn_test_cmd})
 set_tests_properties(
     ${target}
     PROPERTIES
-        LABELS "profile"
+        LABELS "quick;standard;comprehensive;full;pre-commit;smoke;pr;nightly;weekly"
```
Comment (abchoudh-amd) [inline projects/rocprofiler-compute/src/lib/roctx_recordfn/tests/CMakeLists.txt]: The test labels should be picked up from test_categories.yaml based on the test_categories the test is added to.

**Comment by vedithal-amd** [comment]
Pushed [`982999c016`.](https://github.com/ROCm/rocm-systems/pull/9446/commits/982999c016d3d4f1a1c3cce8b0ca45221a30b166)

- Moved `add_test()` to the top-level `CMakeLists.txt`, matching the other two gtests. YAML labels are applied in that scope and `set_property(TEST ...)` is directory-scoped.
- Guarded it with `if(TARGET test-roctx-recordfn)`. The target only exists when torch and the Python dev files are found.
- Dropped the hardcoded `LABELS`. Tiers come from `tests/test_categories.yaml` now, so they cannot drift. Listing a test that may not exist is safe, the generated code guards each name with `if(TEST ...)`.
- Added to `quick`, `standard`, `comprehensive`, `full`. Skipped `sanitizer`, since uninstrumented torch plus a Python interpreter means false positives.
- Exported the test's `LD_LIBRARY_PATH` through the cache so the top-level scope can read it.
- Also fixes install-tree ctest. Only the top-level `CTestTestfile.cmake` is installed, so a subdir `add_test()` never reached it. 53 tests to 54.

Verified locally: `-L full -LE full_rocprofiler_compute_ci_exclude` selects it, and it passes from the install tree.

**Comment by vedithal-amd** [comment]
Pushed [`753ad74d59`.](https://github.com/ROCm/rocm-systems/pull/9446/changes/753ad74d5906c4fdea001227a62d78b647ae0fbf)

- Test now runs through `tests/roctx_recordfn_launcher.py`, which finds the torch, openblas and roctx lib dirs on the machine running it. Same search lists the CMake used, just at test time.
- Dropped the target's `INSTALL_RPATH` and the baked ctest `ENVIRONMENT`. Both held build-machine absolute paths, dead wherever build and test are separate steps.
- Removed the openblas glob and the LD path cache from `src/lib/roctx_recordfn/tests/CMakeLists.txt`, 45 lines gone. What stays in the parent is link-time only.
- Added `SKIP_RETURN_CODE 125`. The test-time interpreter comes from PATH and need not match the one pinned at configure, so torch may be missing there.

Verified: passes from the install tree with `LD_LIBRARY_PATH` stripped, and in a build tree configured without `TEST_FROM_INSTALL`. Installed binary now has no RPATH or RUNPATH, and the test definition has no absolute paths.

**Comment by vedithal-amd** [comment]
@ggottipa-amd one more step before this can merge, needs a PR in `ROCm/TheRock` first.

**File:** `profiler/post_hook_rocprofiler-compute.cmake`

**Change:** add `test-roctx-recordfn` to the `foreach` list.

```cmake
foreach(_test_target test-rocprofiler-compute-tool test-pc-sampling-collector test-roctx-recordfn)
```

**Why:** the hook sets `THEROCK_INSTALL_RPATH_ORIGIN`, which tells TheRock where the binary actually lands. Without it the origin falls back to `<INSTALL_DESTINATION>/bin`, so the `$ORIGIN` entries are computed two levels too shallow and roctx plus the sysdeps do not resolve. The `if(TARGET ...)` guard makes it a no-op when the target is not built.

**Then:** bump the TheRock pin in this repo once that merges. #9056 is the pattern, it updates the pinned `ref:` across the `therock-*.yml` workflows.

Merge order: TheRock PR, then the pin bump, then this PR.

**Review by vedithal-amd — CHANGES_REQUESTED** [review]
Blocking for TheRock side to adjust first

**Comment by vedithal-amd** [comment]
@ggottipa-amd are TheRock side PR and bump TheRock pin in rocm-systems PR merged? If not, then this PR should be blocked until those happen.

File: projects/rocprofiler-compute/tests/torch_trace_collector_launcher.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import os
+import sys
+from pathlib import Path
+
+SKIP_RETURN_CODE = 125
+
+_ROCM_ROOTS = (os.environ.get("ROCM_PATH", ""), "/opt/rocm")
+
+
+def _skip(reason: str) -> int:
```
Comment (abchoudh-amd) [inline projects/rocprofiler-compute/tests/torch_trace_collector_launcher.py]: In this repo, the convention is not to use leading `_` for private functions, etc. Could you remove the `_`s.

File: projects/rocprofiler-compute/tests/torch_trace_collector_launcher.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    print(f"SKIP: {reason}")
+    return SKIP_RETURN_CODE
+
+
+def _torch_lib_dirs() -> list[str]:
+    """Torch link dirs plus the wheel lib dir, which some wheels omit."""
+    import torch
```
Comment (abchoudh-amd) [inline projects/rocprofiler-compute/tests/torch_trace_collector_launcher.py]: Could you make the imports global?

File: projects/rocprofiler-compute/tests/torch_trace_collector_launcher.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    from torch.utils import cpp_extension
+
+    dirs = list(cpp_extension.library_paths())
+    dirs.append(str(Path(torch.__file__).parent / "lib"))
+    return dirs
+
+
+def _first_dir_with(suffixes: tuple[str, ...], pattern: str) -> list[str]:
+    for root in _ROCM_ROOTS:
+        if not root:
```
Comment (abchoudh-amd) [inline projects/rocprofiler-compute/tests/torch_trace_collector_launcher.py]: You could strip empty ROCM_ROOTS before the loop.

**Review by abchoudh-amd — CHANGES_REQUESTED** [review]

**Comment by vedithal-amd** [comment]
@ggottipa-amd we dont need this PR - the torch trace collector gtest is already registered in our ctest suite as part of https://github.com/ROCm/rocm-systems/pull/8552

**Review by vedithal-amd — CHANGES_REQUESTED** [review]
we dont need this

**Comment by ggottipa-amd** [comment]
Closing against PR 8552
